runtimes/cloudformation/cfnpatcher/template.go

```diff
@@ -10,102 +10,64 @@ import (
 	"github.com/sysdiglabs/agent-kilt/pkg/kilt"
 )
 
-type TemplateInfo struct {
-	TargetInfo *kilt.TargetInfo
-	// Containers are not null when template values are complex
-	Name                 *gabs.Container
-	Image                *gabs.Container
-	EntryPoint           []*gabs.Container
-	Command              []*gabs.Container
-	EnvironmentVariables map[string]*gabs.Container
-}
-
-func GetValueFromTemplate(what *gabs.Container) (string, *gabs.Container) {
-	var result string
-	var fallback *gabs.Container
-
-	switch v := what.Data().(type) {
-	case string:
-		result = v
-		fallback = nil
-	default:
-		result = "placeholder: " + what.String()
-		fallback = what
-	}
-	return result, fallback
-}
-
-func extractContainerInfo(ctx context.Context, group *gabs.Container, groupName string, container, parameters *gabs.Container, configuration *Configuration) *TemplateInfo {
-	cfnInfo := new(TemplateInfo)
+func extractContainerInfo(ctx context.Context, group *gabs.Container, groupName string, container, parameters *gabs.Container, configuration *Configuration) *kilt.TargetInfo {
 	info := new(kilt.TargetInfo)
-	cfnInfo.TargetInfo = info
 	l := log.Ctx(ctx)
 
-	info.ContainerName, cfnInfo.Name = GetValueFromTemplate(container.S("Name"))
+	info.ContainerName = container.S("Name")
 	info.ContainerGroupName = groupName
-	info.EnvironmentVariables = make(map[string]string)
-	cfnInfo.EnvironmentVariables = make(map[string]*gabs.Container)
+	info.EnvironmentVariables = make(map[string]*gabs.Container)
 
 	if container.Exists("Image") {
-		info.Image, cfnInfo.Image = GetValueFromTemplate(container.S("Image"))
-		if cfnInfo.Image != nil {
-			l.Info().Str("image", info.Image).Msg("retrieving image from template parameters")
+		info.Image = container.S("Image")
+		if info.Image.Exists("Ref") {
+			l.Info().Str("image", info.Image.String()).Msg("retrieving image from template parameters")
 
-			parameterName, ok := cfnInfo.Image.S("Ref").Data().(string)
+			parameterName, ok := info.Image.S("Ref").Data().(string)
 			if ok {
 				image, ok := parameters.S(parameterName).Data().(string)
 				if ok {
-					l.Info().Str("image", info.Image).Msgf("found image %s", image)
-					info.Image = image
+					l.Info().Str("image", info.Image.String()).Msgf("found image %s", image)
+					info.Image.Set(image)
 				} else {
-					l.Warn().Str("image", info.Image).Msg("could not resolve the image parameter")
+					l.Warn().Str("image", info.Image.String()).Msg("could not resolve the image parameter")
 				}
 			} else {
-				l.Warn().Str("image", info.Image).Msg("could not find the name of the image parameter")
+				l.Warn().Str("image", info.Image.String()).Msg("could not find the name of the image parameter")
 			}
 		}
 
 		os.Setenv("HOME", "/tmp") // crane requires $HOME variable
-		repoInfo, err := GetConfigFromRepository(info.Image)
+		repoInfo, err := GetConfigFromRepository(info.Image.String())
```

Comment:
Note: this is broken (.String() renders the inner string value with quotes, so we're trying to get e.g. `"quay.io/foo/bar"` instead of `quay.io/foo/bar`). It's fixed in a follow up PR, I only just noticed this.